retrieval/raw/raw_endpoint.go

```diff
@@ -43,19 +49,38 @@ func (c *Client) createRequest(method, url string, body io.Reader) (*http.Reques
 }
 
 //GetRawEndPoint retrives the body of HTTP response from a given sourcename, working directory, and URL
-func (c *Client) GetRawEndPoint(sourceName string, workDir *os.File, URL string) (rawRespFile *os.File, rerr error) {
+func (c *Client) GetRawEndPoint(sourceName string,
+	workDir *os.File, URL string) (rawRespFile *os.File, err error) {
+
+	attempts := c.retries + 1
+
+	for i := uint(0); i < attempts; i++ {
+		rawRespFile, err = downloadToFile(c, sourceName, workDir, URL, i)
+		if err != nil {
+			log.Printf("%v URL: %s retrying: %v", err, URL, i)
+			time.Sleep(time.Duration(int64(math.Pow(2, float64(i)))) * time.Second)
+			continue
+		}
+		return rawRespFile, err
+	}
+	return nil, err
+}
+
+func downloadToFile(c *Client, sourceName string,
+	workDir *os.File, URL string, retryCount uint) (rawRespFile *os.File, rerr error) {
 
-	var empty os.File
 	var fileExt string
 
 	req, err := c.createRequest("GET", URL, nil)
 	if err != nil {
-		return &empty, errors.New("Unable to create raw request for " + sourceName)
+		return rawRespFile, errors.New("Unable to create raw request for " + sourceName)
 	}
 
 	resp, err := c.HTTPClient.Do(req)
 	if err != nil {
-		return &empty, err
+		return rawRespFile, errors.New("Unable to connect ")
+	} else if !(resp.StatusCode >= 200 && resp.StatusCode <= 299) {
+		return rawRespFile, fmt.Errorf("Invalid response %s", strconv.Itoa(resp.StatusCode))
 	}
 
 	defer util.SafeClose(resp.Body.Close, &rerr)
```

Comment:
still want to move this defer to probably before the `else if` block. maybe breaking the if/else if into just 2 if blocks, with the defer in between